File: libraries/cyclestreets-view/src/main/java/net/cyclestreets/liveride/Arrivee.java
package net.cyclestreets.liveride;

import net.cyclestreets.routing.Journey;

import org.osmdroid.util.GeoPoint;

final class Arrivee extends LiveRideState
{
  public static final String ARRIVEE = "Arrivée";

  Arrivee(final LiveRideState previous) {
    super(previous);
    notify(ARRIVEE, ARRIVEE);
  }

  @Override
  public LiveRideState update(Journey journey, GeoPoint whereIam, int accuracy) {
    return new Stopped(getContext());
  }

  @Override
  public boolean isStopped() { return false; }
  @Override
  public boolean arePedalling() { return false; }
}

File: libraries/cyclestreets-view/src/main/java/net/cyclestreets/liveride/Arrivee.kt
package net.cyclestreets.liveride

import android.util.Log
import net.cyclestreets.routing.Journey
import org.osmdroid.util.GeoPoint

internal class Arrivee(previous: LiveRideState) : LiveRideState(previous) {

    companion object {
        const val ARRIVEE = "Arrivée"
    }

    init {
        notify(ARRIVEE, important = true)
    }

    override fun update(journey: Journey, myLocation: GeoPoint, accuracy: Int): LiveRideState {
        return Stopped(context)
    }

    override fun isStopped(): Boolean { return false }
    override fun arePedalling(): Boolean { return false }
}